pkg/webhook/validator.go

```diff
@@ -382,55 +384,56 @@ func ValidatePolicy(ctx context.Context, namespace string, ref name.Reference, c
 			switch {
 			case authority.Static != nil:
 				if authority.Static.Action == "fail" {
-					result.err = errors.New("disallowed by static policy")
+					result.err = cosign.NewVerificationError("disallowed by static policy")
 					results <- result
 					return
 				}
-				result.signatures = []PolicySignature{{Subject: "allowed by static policy", Issuer: "allowed by static policy"}}
+
 			case len(authority.Attestations) > 0:
 				// We're doing the verify-attestations path, so validate (.att)
-				validatedAttestations, err := ValidatePolicyAttestationsForAuthority(ctx, ref, authority, authorityRemoteOpts...)
-				if err != nil {
-					result.err = err
-				} else {
-					result.attestations = validatedAttestations
-				}
+				result.attestations, result.err = ValidatePolicyAttestationsForAuthority(ctx, ref, authority, authorityRemoteOpts...)
+
 			default:
-				validatedSignatures, err := ValidatePolicySignaturesForAuthority(ctx, ref, authority, authorityRemoteOpts...)
-				if err != nil {
-					result.err = err
-				} else {
-					result.signatures = validatedSignatures
-				}
+				result.signatures, result.err = ValidatePolicySignaturesForAuthority(ctx, ref, authority, authorityRemoteOpts...)
 			}
 			results <- result
 		}()
 	}
+
 	// If none of the Authorities for a given policy pass the checks, gather
 	// the errors here. Even if there are errors, return the matched
 	// authoritypolicies.
-	authorityErrors := []error{}
+	authorityErrors := make([]error, 0, len(cip.Authorities))
 	// We collect all the successfully satisfied Authorities into this and
 	// return it.
-	policyResult := &PolicyResult{AuthorityMatches: make(map[string]AuthorityMatch)}
-	for i := 0; i < len(cip.Authorities); i++ {
+	policyResult := &PolicyResult{
+		AuthorityMatches: make(map[string]AuthorityMatch, len(cip.Authorities)),
+	}
+	for range cip.Authorities {
 		select {
 		case <-ctx.Done():
-			authorityErrors = append(authorityErrors, fmt.Errorf("context was canceled before validation completed"))
+			authorityErrors = append(authorityErrors, fmt.Errorf("%w before validation completed", ctx.Err()))
+
 		case result, ok := <-results:
 			if !ok {
-				authorityErrors = append(authorityErrors, fmt.Errorf("results channel failed to produce a result"))
+				authorityErrors = append(authorityErrors, errors.New("results channel closed before all results were sent"))
```

Comment:
just for my own edification, is that really the only reason why that would return !ok?

Reply:
Yup:
<img width="717" alt="image" src="https://user-images.githubusercontent.com/2442466/181944128-a57c1a7e-2994-412c-ad84-8b6213d181ca.png">

Reply:
Thanks lazygithub :)